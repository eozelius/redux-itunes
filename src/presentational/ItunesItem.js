import React from 'react'
import PropTypes from 'prop-types'
import ItuneItemStyle from '../styles/ItunesItem.module.css'

const ItunesItem = props => {
  const image = props.data.artworkUrl100 
    ? <img src={props.data.artworkUrl100} alt='album artwork' />
    : <p>album artwork not available</p>
  
  return (
    <div className={`${ItuneItemStyle.itunesItem}`} >
      <h2>{props.data.artistName}</h2>
      <label>{props.data.collectionName}</label> <br/>
      { image }
      <p>{props.data.trackName}</p>
    </div>
  )
}

ItunesItem.propTypes = {
  data: PropTypes.object
}

ItunesItem.defaultProps = {
  data: {
    type: '',
    artistName: '',
    trackName: ''
  }
}

export default ItunesItem
