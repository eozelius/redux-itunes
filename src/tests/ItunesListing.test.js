import ItunesListing from '../presentational/ItunesListings'
import ItunesItem from '../presentational/ItunesItem'
import React from 'react'

describe(ItunesListing, () => {
  test('it renders properly', () => {
    const listing = shallow(<ItunesListing />)
    expect(listing).toMatchSnapshot()
  })

  test('it presents a "please search..." <label> if no listings are passed', () => {
    const listing = shallow(<ItunesListing />)
    expect(listing.find('.loading').length).toEqual(1)
  })

  test('it renders a list of Itunes Items', () => {
    const mockListings = [{}, {}, {}]
    const listing = mount(<ItunesListing listings={mockListings} />)
    expect(listing.find(ItunesItem).length).toEqual(3)
  })
})
