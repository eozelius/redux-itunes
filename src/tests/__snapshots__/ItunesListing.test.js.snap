// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItunesListings it renders properly 1`] = `
<div
  className="wrapper"
>
  <label
    className="loading"
  >
    Search for an artist or song...
  </label>
</div>
`;
